rclcpp/include/rclcpp/node.hpp

```diff
@@ -142,76 +142,27 @@ class Node : public std::enable_shared_from_this<Node>
   /// Create and return a Publisher.
   /**
    * \param[in] topic_name The topic for this publisher to publish on.
-   * \param[in] qos_history_depth The depth of the publisher message queue.
-   * \param[in] allocator Optional custom allocator.
+   * \param[in] group The callback group for this publisher. NULL for no callback group.
+   * \param[in] options Additional options to control creation of the publisher.
    * \return Shared pointer to the created publisher.
    */
   template<
-    typename MessageT, typename Alloc = std::allocator<void>,
-    typename PublisherT = ::rclcpp::Publisher<MessageT, Alloc>>
-  std::shared_ptr<PublisherT>
-  create_publisher(
-    const std::string & topic_name, size_t qos_history_depth,
-    std::shared_ptr<Alloc> allocator = nullptr);
-
-  /// Create and return a Publisher.
-  /**
-   * \param[in] topic_name The topic for this publisher to publish on.
-   * \param[in] qos_profile The quality of service profile to pass on to the rmw implementation.
-   * \param[in] allocator Optional custom allocator.
-   * \return Shared pointer to the created publisher.
-   */
-  template<
-    typename MessageT, typename Alloc = std::allocator<void>,
+    typename MessageT,
+    typename Alloc = std::allocator<void>,
     typename PublisherT = ::rclcpp::Publisher<MessageT, Alloc>>
   std::shared_ptr<PublisherT>
```

Comment:
reminder: restore pre-existing `create_publisher` and `create_subscriber` APIs, and mark as deprecated

Reply:
@emersonknapp Will you be carving out the `PublisherOptions`/`SubscriptionOptions` PR? Can you restore the to-be-deprecated APIs as part of that? I want to leave them out here so that uses of the to-be-deprecated APIs will error out instead of just producing some warnings.

Reply:
Yes, I am working on that today - once that's ready we can rebase this review onto that branch